src/kafka/consumer.cr

```diff
@@ -74,7 +74,9 @@ module Kafka
     # :nodoc:
     def finalize
       LibC.free(@p_err_str)
-      LibKafkaC.topic_partition_list_destroy(@topics) if @topics
+      if topics = @topics
+        LibKafkaC.topic_partition_list_destroy(topics)
```

Comment:
I see you use the `@topics.try {}` style in the producer, I'll update to use that